client/document/desktop.jsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React, { Fragment } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { jsonStringifyForHtml } from '../../server/sanitize';
+import Head from '../components/head';
+import getStylesheet from './utils/stylesheet';
+
+class Desktop extends React.Component {
+	render() {
+		const {
+			app,
+			faviconURL,
+			i18nLocaleScript,
+			isRTL,
+			lang,
+			urls,
+			hasSecondary,
+			config,
+			isFluidWidth,
+			env,
+			isDebug,
+			badge,
+			abTestHelper,
+			branchName,
+			commitChecksum,
+			devDocs,
+			devDocsURL,
+			feedbackURL,
+		} = this.props;
+		return (
+			<html
+				lang={ lang }
+				dir={ isRTL ? 'rtl' : 'ltr' }
+				className={ classNames( 'is-desktop', { 'is-fluid-width': isFluidWidth } ) }
+			>
+				<Head title="WordPress.com" faviconURL={ faviconURL } cdn={ '//s1.wp.com' }>
+					<link
+						rel="stylesheet"
+						id="main-css"
+						href={
+							urls[ getStylesheet( { rtl: !! isRTL, debug: isDebug || env === 'development' } ) ]
+						}
+						type="text/css"
+					/>
+					<link rel="stylesheet" id="desktop-css" href="/desktop/wordpress-desktop.css" />
+				</Head>
+				<body className={ classNames( { rtl: isRTL } ) }>
+					{ /* eslint-disable wpcalypso/jsx-classname-namespace, react/no-danger */ }
+					<div id="wpcom" className="wpcom-site">
+						<div className="layout">
+							<div className="masterbar" />
+							<div className="layout__content">
+								<div className="wpcom-site__logo noticon noticon-wordpress" />
+								{ hasSecondary && (
+									<Fragment>
+										<div className="layout__secondary" />
+										<ul className="sidebar" />
+									</Fragment>
+								) }
+							</div>
+						</div>
+					</div>
+					{ badge && (
+						<div className="environment-badge">
+							{ abTestHelper && <div className="environment is-tests" /> }
+							{ branchName &&
+								branchName !== 'master' && (
+									<span className="environment branch-name" title={ 'Commit ' + commitChecksum }>
+										{ branchName }
+									</span>
+								) }
+							{ devDocs && (
+								<span className="environment is-docs">
+									<a href={ devDocsURL } title="DevDocs">
+										docs
+									</a>
+								</span>
+							) }
+							<span className={ `environment is-${ badge } is-env` }>{ badge }</span>
+							<a
+								className="bug-report"
+								href={ feedbackURL }
+								title="Report an issue"
+								target="_blank"
+							/>
+						</div>
+					) }
+
+					{ app && (
+						<script
+							type="text/javascript"
+							dangerouslySetInnerHTML={ {
+								__html: 'var app = ' + jsonStringifyForHtml( app ),
+							} }
+						/>
+					) }
+					{ config && (
```

Comment:
this question is coming from a place of ignorance.
instead of `dangerouslySetInnerHtml`, could we write:
```jsx
{ config && <script type="text/javascript"> { config } </script> }
```

Reply:
I was hoping you would apply it to both `dengerouslySetInnerHtml`s 😰

Reply:
@samouri it seems to me that this doesn't work as expected. If I don't set the vars via `dangerouslySetInnerHTML` the output on the client looks like this:

<img width="726" alt="screen shot 2018-02-16 at 21 47 01" src="https://user-images.githubusercontent.com/9202899/36328680-6aed3ac6-1363-11e8-9d76-a199098c2540.png">

On the other hand via `dangerouslySetInnerHTML`:

<img width="546" alt="screen shot 2018-02-16 at 21 47 18" src="https://user-images.githubusercontent.com/9202899/36328689-78e27182-1363-11e8-9b1a-65ecc6d2cb55.png">

If you know a way around this I'm happy to learn something new :) for now I reverted the changes back to using `dangerouslySetInnerHTML`.

Reply:
I'm okay with shipping this as-is. I have a couple theories on how to improve this a bit in a future PR.